### ISSUE TITLE
dispatch action will be missed when take(actionCreator)

### ISSUE THREAD

**Issue by luhui** [jfairbank/redux-saga-test-plan#165]
Test with sample code below,
```js
//@flow

import { expectSaga } from 'redux-saga-test-plan'
import { take, call } from 'redux-saga/effects'
import { createAction } from 'redux-actions'

const actionA = createAction('a')
const actionB = createAction('b')

function* testSaga1(fn) {
  while (true) {
    yield take(actionA)
    yield call(fn)
  }
}

function* testSaga2(fn) {
  while (true) {
    yield take(actionA.toString())
    yield call(fn)
  }
}

it('dispatch action test with testSaga1', () => {
//expect 3 calls, but get 2 calls
  return runTest(testSaga1)
})

it('dispatch action test with testSaga2', () => {
  return runTest(testSaga2)
})

function runTest(saga) {
  const fn = jest.fn()
  return expectSaga(saga, fn)
          .dispatch(actionA())
          .dispatch(actionA())
          .dispatch(actionB()) // if we remove this line, testSaga1 will be passed
          .dispatch(actionA())
          .run()
          .then(() => expect(fn).toHaveBeenCalledTimes(3))
}

```

testSaga1 is failed, the third actionA will be omit (and if you move actionB below the first actionA, and the other actionA will be omit).
testSaga2 is passed.

The difference between testSaga1 and testSaga2 is that testSaga1 take an action creator, while the testSaga2 take a string.

**Comment by jfairbank** [comment]
Thanks for the bug report and test case @luhui!

It looks like redux-saga-test-plan isn't checking for the `toString` property on the action creator like redux-saga does. I'll have a fix out for this soon.

**Comment by jfairbank** [comment]
Fixed in v3.4.0. Please let me know if you have issues.
